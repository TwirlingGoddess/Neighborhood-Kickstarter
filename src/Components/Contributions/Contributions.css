.project-comment-section {
  display: flex;
  height: 100vh;
  width: 100%;
}

.project-section {
  width: 35%;
  height: 90vh;
  background-image: radial-gradient(to top, #09203f 0%, #537895 100%);
  margin: 1rem 3rem;
  color: white;
  font-size: x-large;
  overflow: scroll;
  border-radius: 0.7em;

}

.comment-section {
  margin: 1rem 3rem;
  height: 95vh;
  display: flex;
  justify-content: space-around;
  flex-direction: column-reverse;
  width: 65%;
  border-radius: 0.7em;

}

.add-comment-input {
  border-radius: 0.7em;
  font-size: 2rem;
  height: 10rem;
  width: 82.5%;
  background-color: transparent;
  border: 1px solid white;
  color: white
}

.comment-box {
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  flex-direction: column;
  padding: 1rem;
  width: 79%;
  background-repeat: no-repeat;
  height: 30rem;
  overflow: scroll;
  border: 1px solid white;
  margin-bottom: 1rem;
  border-radius: 0.7em;

}

.comment,
.comment-owner {
  color: white;
  margin-bottom: .2rem
}

.comment-div {
  display: flex;
  width: 100%;
  flex-direction: column;
  align-items: flex-start;
  border-bottom: 1px solid #fff;
  
}

.selected-image {
  width: 27.5rem;
  border-radius: 0.7em 0.7em 0 0;
  height: 20rem;
}

.submit-comment {
  box-sizing: border-box;
  appearance: none;
  background-color: transparent;
  border-radius: 0.6em;
  cursor: pointer;
  align-self: center;
  font-size: 1rem;
  font-weight: 400;
  line-height: 1;
  margin: 20px;
  padding: 1em 1em;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-family: 'Montserrat', sans-serif;
  font-weight: 700;
  border-color: #3498db;
  color: #fff;
  box-shadow: 0 0 40px 40px #3498db inset, 0 0 0 0 #3498db;
  transition: all 150ms ease-in-out;
  width: 20rem;
  max-height: 4rem;
  margin-right: 10rem;
}

.comment-header {
  color: white;
}

.submit-comment:hover {
  color: #fff;
  outline: 0;
  box-shadow: 0 0 10px 0 #3498db inset, 0 0 10px 4px #3498db;
}