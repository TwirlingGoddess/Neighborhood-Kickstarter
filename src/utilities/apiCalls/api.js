var fs = require('fs');
var S3FS = require('s3fs');
var s3fsImpl = new S3FS('leeleestestbucket1', {
  accessKeyId: 'AKIAIXU4Q4Y4XZWUK4SQ', 
  secretAccessKey: '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'
});

s3fsImpl.create();

var multiparty = require('connect-multiparty'),
  multipartyMiddleware = multiparty();

module.exports = function(router, passport){
  router.use(multipartyMiddleware);

  router.use(passport.authenticate('bearer', { session: false}));
  router.use(function(req, res, next){
    fs.appendFile('log.txt' req.path + " token: " + req.query.access_token + " ")
      function(err){
        next();
      });
  });

  router.post('/testupload', function(req, res){
    var file = req.files.file;
    var stream = fs.createReadStream(file.path);
    return s3fsImpl.writeFile(file.originalFilename, stream).then(function(){
      fs.unlink(file.path, function(err){
        if(err){
          console.error(err)
        }
      })
      res.redirect('/profile');
    })
  })

  router.get('/testAPI', function(req, res){
    res.json({ SecretData: 'abc123' });
  });


}