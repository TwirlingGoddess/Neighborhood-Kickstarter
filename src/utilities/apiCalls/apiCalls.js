export const getNeighborhoods = async () => {
  const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/neighborhoods');
  const neighborhoods = await response.json();
  return neighborhoods
}

export const getProjects = async () => {
  const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/projects');
  const projects = await response.json();
  return projects
}

export const getNeighborhoodProjectsById = async (id) => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/neighborhoods/${id}`);
  const neighborhoodProjects = await response.json();
  return neighborhoodProjects
}

export const getUserProjects = async (userId) => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users/${userId}/projects`);
  const userProjects = await response.json();
  return userProjects
}

export const getAllUsers = async () => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users`);
  const users = await response.json();
  return users
}

export const getProjectComments = async (id) => {
  const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/projects/${id}/comments`)
  const comments = response.json();
  return comments
}

export const addNewUser = async (user) => {
  console.log(JSON.stringify(user))
  try {
    const response = await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/google_users`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(user)
    })
    const currentUser = await response.json();
    return currentUser
  } catch (error) {
    throw new Error(error.message)
  }
}

export const addNewUserLocal = async (user) => {
  console.log(JSON.stringify(user))
  try {
    const response = await fetch('https://guarded-garden-68388.herokuapp.com/api/v1/users', {
      method: 'POST',
      headers: {
        'Content-type': 'application/json',
        'Accept': 'application/json'
      },
      body: JSON.stringify(user)
    })
    const currentUser = await response.json();
    return currentUser
  } catch (error) {
    throw new Error(error.message)
  }
}

export const postNewProject = async (newProject, id) => {
  try {
    await fetch(`https://guarded-garden-68388.herokuapp.com/api/v1/users/${id}/projects`, {
      method: 'POST',
      headers: {
        'Content-type': 'application/json'
      },
      body: JSON.stringify(newProject)
    })
  } catch (error) {
    throw new Error(error.message);
  }
}