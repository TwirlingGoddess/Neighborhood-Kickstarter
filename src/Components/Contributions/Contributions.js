import React, { Component } from 'react';
import { getProjectsComments, postNewComment } from '../../utilities/apiCalls/apiCalls';
import './Contributions.css';
import PropTypes from 'prop-types';


class Contributions extends Component {
  constructor() {
    super();
    this.state = {
      comment: '',
      selectedProject: {},
      projectComments: []
    };
  }

  componentDidMount = () => {
    this.setCurrentProject();
  }

  setCurrentProject = () => {
    if (this.props.currentProject.id) {
      let id = this.props.currentProject.id;
      this.setState({
        selectedProject: this.props.currentProject
      }, () => this.getProjectComments(id));
    }
    return; 
  }

  getProjectComments = async () => {
    if (this.state.selectedProject.id) {
      let projectId = this.state.selectedProject.id;
  
      let projectComments = await getProjectsComments(projectId);

      this.setState({
        projectComments
      });
    }
    return; 
  }

  handleChange = (event) => {
    let value = event.target.value;

    this.setState({
      comment: value
    });
  }

  handleSubmit = async () => {
    let user_id = this.props.currentUser.id;
    let projectId = this.props.currentProject.id;
    let newComment = {content: this.state.comment, user_id};

    await postNewComment(newComment, projectId);
    await this.getProjectComments();
    this.setState({
      comment: ''
    });
  }



  render () {
    let { contact, description, id, photo, resources, title, neighbor } = this.state.selectedProject;
    if (id) {
      return (
        <div className='project-comment-section'>
          <div className='project-section'>
            <img className='selected-image' src={photo} alt='view if the project'/>
            <h2>title: {title}</h2>
            <h2>owner: {neighbor}</h2>
            <h2>contact: {contact}</h2>
            <h2>description: {description}</h2>
            <h2>needs:</h2>
            {resources.map((resource, index) => {
              return <h3 key={index}>{resource.name}</h3>;
            })}
          </div>
          <div className='comment-section'>
            <button onClick={this.handleSubmit} className='submit-comment'>Submit Comment</button>
            <textarea
              className='add-comment-input'
              type='text'
              onChange={this.handleChange}
              placeholder='leave a comment'
              name='comment'
              value={this.state.comment}
            />
            <div className='comment-box'>
              <h1 className='comment-header'>Comments</h1>
              {this.state.projectComments.map((comment, index) => {
                return <div className='comment-div' key={index}>
                  <h2 className='comment-owner'>{comment.author} said:</h2>
                  <h2 className='comment'>{comment.content}</h2>
                </div>;
              })}
            </div>
          </div>
        </div>
      );
    } else {
      return (
        <div>
          <h1>no selected project</h1>
        </div>
      );
    }
  }
}

Contributions.propTypes = {
  currentProject: PropTypes.object,
  currentUser: PropTypes.object
};

export default Contributions;
