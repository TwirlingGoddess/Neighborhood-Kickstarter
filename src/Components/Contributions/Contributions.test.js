import React from 'react';
import Contributions from './Contributions';
import { getProjectsComments, postNewComment } from '../../utilities/apiCalls/apiCalls';
import { shallow } from 'enzyme';

describe('Contributions', () => {
  let wrapper;
  let mockCurrentUser = {};
  let mockCurrentProject = {};

  it('should match the snapshot', () => {
    wrapper = shallow(<Contributions currentUser={mockCurrentUser} currentProject={mockCurrentProject}/>);
    expect(wrapper).toMatchSnapshot();
  });

  it('should update state if there is a current project and it has an id', () => {
    mockCurrentProject = {id: 1, resources: []};
    wrapper = shallow(<Contributions currentUser={mockCurrentUser} currentProject={mockCurrentProject} />);
    wrapper.instance().setCurrentProject();
    expect(wrapper.state('selectedProject')).toEqual(mockCurrentProject);
  });

  it('should not update state if there is no current project id', () => {
    mockCurrentProject = {};
    wrapper = shallow(<Contributions currentUser={mockCurrentUser} currentProject={mockCurrentProject} />);
    wrapper.instance().setCurrentProject();
    expect(wrapper.state('selectedProject')).toEqual({});
  });

  it('should update state when a letter is pressed to add a comment', () => {
    let eventObject = { target: { name: 'comment', value: 'a' } };
    wrapper = shallow(<Contributions currentUser={mockCurrentUser} currentProject={mockCurrentProject} />);
    wrapper.instance().handleChange(eventObject);
    expect(wrapper.state('comment')).toEqual('a');
  });

  it('should return data from fetch', async () => {
    let data = await postNewComment();
    expect(data).toBe(undefined);
  })

  
}); 