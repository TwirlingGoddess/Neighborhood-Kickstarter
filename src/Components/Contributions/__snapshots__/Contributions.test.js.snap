// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contributions should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Contributions
    currentProject={Object {}}
    currentUser={Object {}}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h1>
        no selected project
      </h1>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "no selected project",
      },
      "ref": null,
      "rendered": "no selected project",
      "type": "h1",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h1>
          no selected project
        </h1>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "no selected project",
        },
        "ref": null,
        "rendered": "no selected project",
        "type": "h1",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
